### PR TITLE
MS graph mail enhancement

### PR DESCRIPTION
**Pull request by reutshal**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/21307

## Description
- Added 4 commands, which require to add the *Mail.Send Application* permission.
    - ***msgraph-mail-create-draft***
    - ***msgraph-mail-send-draft***
    - ***msgraph-mail-reply-ro***
    - ***send-mail***
- Added the ability to ***fetch incidents***.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [ ] Tests
- [ ] Documentation 
- [X] Code Review

### CODE REVIEW

**Comment by reutshal** [comment]
@kirbles19 Can you please go over the files:
- https://github.com/demisto/content/blob/MS_graph_mail-enhancement/Integrations/MicrosoftGraphMail/CHANGELOG.md
- https://github.com/demisto/content/blob/MS_graph_mail-enhancement/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.yml
- https://github.com/demisto/content/blob/MS_graph_mail-enhancement/Integrations/MicrosoftGraphMail/MicrosoftGraphMail_description.md
We want to merge this PR before the code freeze on Sunday morning :)
Thanks!

File: Integrations/MicrosoftGraphMail/MicrosoftGraphMail_description.md [hunk 1]
```diff
@@ -1,2 +1,15 @@
...
+- ***msgraph-mail-create-draft***
+- ***msgraph-mail-send-draft***
+- ***msgraph-mail-reply-ro***
+- ***send-mail***
+
+To use these commands and to fetch incidents,
+you will need to add to your application the *Mail.Send application* permission (not the delegated one),
+and re-authorize your integration's instance.
+
+If you do not wish to use these commands, you may keep your integration credentials the same.
```
Comment (ronykoz) [inline Integrations/MicrosoftGraphMail/MicrosoftGraphMail_description.md]: you need to add this to the CHANGELOG

File: Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py [hunk 2]
```diff
@@ -626,6 +1289,134 @@ def get_email_as_eml_command(client: MsGraphClient, args):
...
+    Sends email from user's mailbox, the sent message will appear in Sent Items folder
+    """
+    prepared_args = prepare_args('send-mail', args)
+    email = args.get('from', client._mailbox_to_fetch)
+    suffix_endpoint = f'/users/{email}/sendMail'
+    message_content = MsGraphClient.build_message(**prepared_args)
+    client.ms_client.http_request('POST', suffix_endpoint, json_data={'message': message_content}, resp_type="text")
+
+    message_content.pop('attachments', None)
+    message_content.pop('internet_message_headers', None)
```
Comment (ronykoz) [inline Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py]: why are we removing it, we should have the names in the context

File: Integrations/MicrosoftGraphMail/MicrosoftGraphMail_description.md (hunk 1 above)
Comment (reutshal) [reply Integrations/MicrosoftGraphMail/MicrosoftGraphMail_description.md]: Done

File: Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py (hunk 2 above)
Comment (reutshal) [reply Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py]: Agreed on keeping it

**Pull request merged**